### PR TITLE
fix: updating binary printing logic

### LINKED ISSUE #153: Indent complex conditions
Input:
```php
                        if (
                            $showNoticeValue === '1' &&
                            !is_null($currentService) &&
                            $currentService->getServiceId() === $activation->getServiceId()
                        ) {
```
Output:
```php
                        if (
                            $showNoticeValue === '1' &&
                            !is_null($currentService) &&
                            $currentService->getServiceId() ===
                            $activation->getServiceId()
                        ) {
```
I'd suggest
```php
                        if (
                            $showNoticeValue === '1' &&
                            !is_null($currentService) &&
                            $currentService->getServiceId() ===
                                $activation->getServiceId()
                        ) {
```

### LINKED ISSUE #231: Some bin operation not indent
Input:
```
'tesssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssst' . 'test';
```
Output:
```
'tesssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssst' 
. 'test';
```

It is entailed an additional logic for indenting in some structures. We need solve this and remove extra logic :+1:

### PR DESCRIPTION
**Pull request by mgrip**
fixes #153 fixes #231

### CODE REVIEW

File: tests/comments/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -163,17 +163,16 @@ $variable =
...
     (
         $this->modelData instanceof \\Traversable &&
-        0 === iterator_count($this->modelData)
+            0 === iterator_count($this->modelData)
     );
 
 if (
-    2 & /* MB_OVERLOAD_STRING */
-    (int) ini_get('mbstring.func_overload')
+    2 & (int) ini_get('mbstring.func_overload') /* MB_OVERLOAD_STRING */
```
Comment (mgrip) [inline tests/comments/__snapshots__/jsfmt.spec.js.snap]: I couldn't seem to figure this one out - I think there enough improvements though its worth pushing this and opening up an issue for this case with the comment

Comment (czosel) [reply tests/comments/__snapshots__/jsfmt.spec.js.snap]: :+1: agree

File: src/printer.js [hunk 2]
```diff
@@ -639,6 +639,84 @@ function isPropertyLookupChain(node) {
   return isPropertyLookupChain(node.what);
 }
 
+function shouldInlineLogicalExpression(node) {
+  if (node.right.kind === "array" && node.right.items.length !== 0) {
+    return true;
+  }
+  return false;
```
Comment (czosel) [inline src/printer.js]: `return node.right.kind === "array" && node.right.items.length !== 0`

**Comment by mgrip** [comment]
@evilebottnawi another big one - you might want to take a look before we merge

**Comment by czosel** [comment]
:+1: Looks good to me! (I didn't look at the algorithm in detail, but i guess it's mostly the same as in prettier for JS)

**Comment by mgrip** [comment]
good idea - all set 👍

File: tests/operations/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -128,4 +144,25 @@ $a ?? $b;
...
+$someReallyReallyReallyLongBooleanVariable .= $someReallyReallyReallyLongBooleanVariable;
+++$someReallyReallyReallyLongBooleanVariableWithReallyReallyReallyReallyReallyReallyLongName;
+--$someReallyReallyReallyLongBooleanVariableWithReallyReallyReallyReallyReallyReallyLongName;
+$someReallyReallyReallyLongBooleanVariableWithReallyReallyReallyReallyReallyReallyLongName++;
+$someReallyReallyReallyLongBooleanVariableWithReallyReallyReallyReallyReallyReallyLongName--;
+$someReallyReallyReallyLongBooleanVariable +
+    $someReallyReallyReallyLongBooleanVariable -
+    $someReallyReallyReallyLongBooleanVariable;
+
+$value = isset($_POST['menu-item-dropdown']) &&
```
Comment (evilebottnawi) [inline tests/operations/__snapshots__/jsfmt.spec.js.snap]: Related ?https://prettier.io/playground/#N4Igxg9gdgLgprEAuEASAbgQwDYFc4AEAvAQJYDO5cMAFKgPoAKA8gMoAqA2gOQC2CuALSl4vQQBMAThAAO4iAHco3ALoBKAgDJNAHSgEDZStTpM2XPgOGiJ0uYuUrOqflFwBJUQBEARu-HqeoYEAPwE3AiYPthw4txBhkjh4hRRMXEA3CAANCCyMKTQ5MigmJLSCoxlCMUomOgQpOI5IJjkMMgAZjhUuT6SmGAA1tSsMoOkUADmyDCS+LnyYF09cLmTVJIwjANTvJgr2L0gAFbkAB4AQgPDo5j8ADKTcIfHELgwMh8ATK9rIONJJtkCAfFEAJ7YaAtGSSSYwADqTRgAAtkAAOAAMuVhECoCIGMhBsLgm3QL1ykjgAEdcKQqTtMHsDkhukd-lReKRZvMOZMpjEAIq4CDwP65GBRJHiVHIb4SgakbD8gDCEF4+xBUGgFJAuCo7CitTZVAAvqagA

Comment (mgrip) [reply tests/operations/__snapshots__/jsfmt.spec.js.snap]: hrmm interesting - ill try to take a look today

**Comment by mgrip** [comment]
👌 think this should be all set now @evilebottnawi @czosel

**Pull request merged**